src/vaults/SectorCrossVault.sol

```diff
@@ -0,0 +1,284 @@
+	// 	address[] calldata vaults,
+	// 	HarvestSwapParms[] calldata harvestParams
+	// ) public onlyRole(MANAGER) checkInputSize([vaults.length, harvestParams.length]) {
+	// 	for (uint i = 0; i < vaults.length;) {
+	// 		// No idea how to call harvest and how to map swap params with harvest call
+	// 		vaults[i].harvest(harvestParams.min, harvestStrategies.deadline);
+
+	// 		unchecked { i++; }
+	// 	}
+	// }
+
+	// TODO Move modifier to end of file
+	// Solidity type system is not allowing me to use uint[]
+	modifier checkInputSize(uint[3] memory inputSizes) {
+		for (uint i = 1; i < inputSizes.length;) {
+			if (inputSizes[i - 1] != inputSizes[i]) {
+				revert InputSizeNotAppropriate();
+			}
+
+			unchecked { i++; }
+		}
+		_;
+	}
+
+	/* BRIDGE FUNCTIONALLITY */
+
+	/// @notice Struct encoded in Bungee calldata
+	/// @dev Derived from socket registry contract
+	struct MiddlewareRequest {
+		uint256 id;
+		uint256 optionalNativeAmount;
+		address inputToken;
+		bytes data;
+	}
+
+	/// @notice Struct encoded in Bungee calldata
+	/// @dev Derived from socket registry contract
+	struct BridgeRequest {
+		uint256 id;
+		uint256 optionalNativeAmount;
+		address inputToken;
+		bytes data;
+	}
+
+	/// @notice Struct encoded in Bungee calldata
+	/// @dev Derived from socket registry contract
+	struct UserRequest {
+		address receiverAddress;
+		uint256 toChainId;
+		uint256 amount;
+		MiddlewareRequest middlewareRequest;
+		BridgeRequest bridgeRequest;
+	}
+
+	/// @notice Decode the socket request calldata
+	/// @dev Currently not in use due to undertainity in bungee api response
+	/// @param _data Bungee txn calldata
+	/// @return userRequest parsed calldata
+	function decodeSocketRegistryCalldata(bytes memory _data)
+		internal
+		pure
+		returns (UserRequest memory userRequest)
+	{
+		bytes memory callDataWithoutSelector = slice(_data, 4, _data.length - 4);
+		(userRequest) = abi.decode(callDataWithoutSelector, (UserRequest));
+	}
+
+	/// @notice Decodes and verifies socket calldata
+	/// @param _data Bungee txn calldata
+	/// @param _chainId chainId to check in bungee calldata
+	/// @param _inputToken inputWantToken to check in bungee calldata
+	/// @param _receiverAddress receiving address to check in bungee calldata
+	function verifySocketCalldata(
+		bytes memory _data,
+		uint256 _chainId,
+		address _inputToken,
+		address _receiverAddress
+	) internal view {
+		UserRequest memory userRequest;
+		(userRequest) = decodeSocketRegistryCalldata(_data);
+
+		if (userRequest.toChainId != _chainId) {
+			revert("Invalid chainId");
+		}
+		if (userRequest.receiverAddress != _receiverAddress) {
+			revert("Invalid receiver address");
+		}
+		if (userRequest.bridgeRequest.inputToken != _inputToken) {
+			revert("Invalid input token");
+		}
+
+		if (!sectorVaultsWhitelist[_chainId][userRequest.receiverAddress]) revert ReceiverNotWhiteslisted(_receiverAddress);
+	}
+
+	// Who should be responsible for whitelist vaults?
+	// I believe it's the guardian
```

Comment:
It should actually be `onlyOwner` since this is a critical method that could compromise funds